src/transform/src/projection_pushdown.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Push down projections into various operators. TODO.
+
+use std::collections::{HashMap, HashSet};
+
+use expr::{RelationExpr, ScalarExpr};
+use repr::{RelationType, Row};
+
+use crate::TransformArgs;
+
+/// Push down projections TODO
+#[derive(Debug)]
+pub struct ProjectionPushdown;
+
+impl crate::Transform for ProjectionPushdown {
+    fn transform(
+        &self,
+        relation: &mut RelationExpr,
+        _: TransformArgs,
+    ) -> Result<(), crate::TransformError> {
+        self.action(relation, (0..relation.typ().column_types.len()).collect());
+        Ok(())
+    }
+}
+
+impl ProjectionPushdown {
+    /// ...
```

Comment:
Even though it's a WIP, tossing a comment in here would help read it. I'm not certain, for example, what is supposed to happen when we wrap an operator that doesn't support the pushdown. I'm sure I'll discover as I read, but having the intended behavior written down makes that process easier!

Reply:
Added some words here, hope that helps!